common/src/main/java/org/apache/iceberg/common/DynConstructors.java

```diff
@@ -235,6 +236,10 @@ public Builder hiddenImpl(String className, Class<?>... types) {
       throw new RuntimeException("Cannot find constructor for " +
           baseClass + "\n" + formatProblems(problems));
     }
+
+    public Map<String, Throwable> problems() {
+      return ImmutableMap.copyOf(problems);
+    }
```

Comment:
Why does this need to be public? The problems are already added when exceptions are thrown.